sycl/doc/extensions/proposed/sycl_ext_oneapi_usm_properties.asciidoc

```diff
@@ -0,0 +1,173 @@
+= sycl_ext_oneapi_usm_properties
+:source-highlighter: coderay
+:coderay-linenums-mode: table
+
+// This section needs to be after the document title.
+:doctype: book
+:toc2:
+:toc: left
+:encoding: utf-8
+:lang: en
+
+:blank: pass:[ +]
+
+// Set the default source code type in this document to C++,
+// for syntax highlighting purposes.  This is needed because
+// docbook uses c++ and html5 uses cpp.
+:language: {basebackend@docbook:c++:cpp}
+
+== Introduction
+IMPORTANT: This specification is a draft.
+
+NOTE: Khronos(R) is a registered trademark and SYCL(TM) and SPIR(TM) are
+trademarks of The Khronos Group Inc.  OpenCL(TM) is a trademark of Apple Inc.
+used by permission by Khronos.
+
+This extension introduces an alternative way to pass both runtime and compile time properties into USM malloc APIs.
+
+`malloc_device`, `malloc_host`, `malloc_shared` take the properties, pass runtime properties onto runtime libraries and create annotated_ptr base on compile-time properties.
+
+The goal of these changes is to enable information about properties to propagate to the device compiler and thereby enable additional optimization of kernel code. Further, this will also be used by the runtime for allocation in the correct target memory.
```

Comment:
I would drop the second sentence because (1) run time properties can already be passed to the runtime using the existing malloc overloads so something like this can already be accomplished with the existing API and (2) per my earlier comment, I don't think this extension should mention buffer_location.